### PR TITLE
Added UI changes for third-party game creation

### PR DESCRIPTION
**Pull request by blackshadowshade**
Addresses part of #1784.

With this change, it is possible for anyone to create a third-party game from the standard create game page.

@cgolubi1, I haven't added any UI testing of this to the QUnit tests. I'd like some guidance on what sort of test you'd like here.

http://jenkins.buttonweavers.com:8080/job/buttonmen-blackshadowshade/1369/

### CODE REVIEW

File: src/ui/js/Newgame.js [hunk 1]
```diff
@@ -304,7 +353,13 @@ Newgame.formCreateGame = function() {
   var validSelect = true;
   var errorMessage;
 
-  if (!Newgame.activity.playerButton) {
+  if (!Newgame.activity.playerName) {
+    validSelect = false;
+    errorMessage = 'Please select the first player.';
+  } else if (Newgame.activity.playerName == Newgame.activity.opponentName) {
+    validSelect = false;
+    errorMessage = 'The two players cannot be the same person.';
```
Comment (cgolubi1) [inline src/ui/js/Newgame.js]: I would say the two players can't be the same username.  We have no way of knowing whether two players are the same person.  :>)

Comment (blackshadowshade) [reply src/ui/js/Newgame.js]: Ah, good point.  :)

**Comment by cgolubi1** [comment]
I'll get back to your question, but also stand up a dev site for this.

**Comment by cgolubi1** [comment]
54.162.78.2 - irilyth, can you link this up as "ephemeral"?

**Comment by irilyth** [comment]
Done!

**Comment by cgolubi1** [comment]
A bug: when you create a game this way and enter something invalid, the UI reverts back to the first player being you, e.g. in the situation i mentioned in my comments in which you forget to select a button for player one.  The UI should retain all the settings when you try to submit something invalid, and redisplay them all when it lets you try again.

**Comment by blackshadowshade** [comment]
So, I think I've addressed these issues.

http://jenkins.buttonweavers.com:8080/job/buttonmen-blackshadowshade/1372/ (if it passes)

**Comment by cgolubi1** [comment]
Looks like the bug i identified is fixed - when i pick a different player one and choose invalid buttons, i get an error message and the player i chose remains selected as player one.

**Comment by cgolubi1** [comment]
shadowshade: you asked awhile ago about QUnit stuff on this PR.  It looks to me like the `test_Newgame.actionCreateGame_prevvals` test could have an entry added to check that the first player value is retained from previous values (that test would have caught the bug i found, at least in principle).

You should also at the least add a player one check to `test_Newgame.formCreateGame`, and take a quick skim at the `test_Newgame.formCreateGame*` tests to see if there's anywhere else you should be adding an error test.

But those two things are probably sufficient.

**Comment by blackshadowshade** [comment]
QUnit tests added, no changes to the rest of the code. Let me know if you think these tests are okay.

I've rebased this back to one commit.

http://jenkins.buttonweavers.com:8080/job/buttonmen-blackshadowshade/1375/ (if it passes)

**Comment by cgolubi1** [comment]
Okay, this looks good.  Let me convince myself that the rebase is otherwise a noop, then i'll merge.

**Pull request merged**